### DISCUSSION TITLE
Hyperloop iOS not compiling. >SDK 9.2.1 and maybe xCode 14.5

### DISCUSSION THREAD

**Discussion by raybelisle** [appcelerator/titanium_mobile#12822] [Q&A]
For some reason, when I try to compile with hyperloop module enabled in tiapp.xml. This seems to have started when I upgraded to SDK 14.5 because my colleague is still on 14.4 and with the same codebase, can compile without issues. Has something changed with 14.5? We don't actually have any hyperloop integration, but I wanted to test something and can't get it to compile successfully with the module enabled. It fails during xCode compile. Something I can check?

This is the error I am getting with debug set to trace.

`[ERROR] 2021-05-19 12:16:42.004 xcodebuild[40927:7803982]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-18108/DTDeviceKitBase/DTDKRemoteDeviceData.m:371
[ERROR] Details:  (null) deviceType from 9d73c18d952e15026ce345b4e06fcb567ba2bd69 was NULL when -platform called.
[ERROR] Object:   <DTDKMobileDeviceToken: 0x7fd611e63260>
[ERROR] Method:   -platform
[ERROR] Thread:   <NSThread: 0x7fd60e8df210>{number = 4, name = (null)}
[ERROR] Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
[ERROR] 2021-05-19 12:16:42.157 xcodebuild[40927:7803992]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-18108/DTDeviceKitBase/DTDKRemoteDeviceData.m:371
[ERROR] Details:  (null) deviceType from 9d73c18d952e15026ce345b4e06fcb567ba2bd69 was NULL when -platform called.
[ERROR] Object:   <DTDKMobileDeviceToken: 0x7fd611e63260>
[ERROR] Method:   -platform
[ERROR] Thread:   <NSThread: 0x7fd60e933f00>{number = 10, name = (null)}
[ERROR] Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
[ERROR] 2021-05-19 12:16:42.215 xcodebuild[40927:7803992]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-18108/DTDeviceKitBase/DTDKRemoteDeviceData.m:371
[ERROR] Details:  (null) deviceType from 9d73c18d952e15026ce345b4e06fcb567ba2bd69 was NULL when -platform called.
[ERROR] Object:   <DTDKMobileDeviceToken: 0x7fd611e63260>
[ERROR] Method:   -platform
[ERROR] Thread:   <NSThread: 0x7fd60e933f00>{number = 10, name = (null)}
[ERROR] Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
[ERROR] ** BUILD FAILED **
[ERROR] The following build commands failed:
[ERROR]         Ld /Users/raybelisle/Documents/app/build/iphone/build/Intermediates/app.build/Debug-iphoneos/app.build/Objects-normal/armv7/Binary/app normal armv7
[ERROR] (1 failure)`

**Comment by Max87ZA** [comment]
I have this issue for a long time, I think it's not iOS/Xcode version related.
 I fixed it by deleting Hyperloop 7.0.1 installed by newest SDK. Even SDK 10.0 can use older version of hyperloop. It's here: /Users/<username>/Library/Application Support/Titanium/modules/iphone/hyperloop. Currently I use 6.0.2 without issues

**Reply by raybelisle** [reply]
I am able to get it to work with the latest version of the Ti SDK. V10. Something to try though. Thanks
